oauth.Dockerfile

```diff
@@ -0,0 +1,45 @@
+# Build the manager binary
+FROM golang:1.18 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+COPY static/ static/
+
+# Copy the go source
+COPY cmd/ cmd/
+COPY api/ api/
+COPY controllers/ controllers/
```

Comment:
can it utilize docker cache from operator build? If not, I think we can remove `controllers` here and from original `Dockerfile` remove `static` and `oauth`.